mondrian/src/main/java/mondrian/rolap/RolapResult.java

```diff
@@ -1298,6 +1298,7 @@ private void executeStripe(
                     point.setAxis(axisOrdinal, tupleIndex);
                     final int savepoint = revaluator.savepoint();
                     try {
+                        revaluator.setEvalAxes( true );
```

Comment:
No, I haven't done any performance checks yet.
I am going to write one performance test suggested by you which will cover this situation.
Also I will look in others performance tests, which were already wrote, and figure out how big impact this fix introduce.